tools/conversion.js

```diff
@@ -10,182 +10,211 @@ if (!dir) {
-  // ex. Nan::To<int>(info[0]) to info[0].Int32Value()
-  [/Nan::To<int>\((.+?)\)/g, '$1.As<Napi::Number>().Int32Value()'],
-  // ex. Nan::To<int32_t>(info[0]) to info[0].Int32Value()
-  [/Nan::To<int32_t>\((.+?)\)/g, '$1.As<Napi::Number>().Int32Value()'],
-  // ex. Nan::To<uint32_t>(info[0]) to info[0].Uint32Value()
-  [/Nan::To<uint32_t>\((.+?)\)/g, '$1.As<Napi::Number>().Uint32Value()'],
-  // ex. Nan::To<int64_t>(info[0]) to info[0].Int64Value()
-  [/Nan::To<int64_t>\((.+?)\)/g, '$1.As<Napi::Number>().Int64Value()'],
-  // ex. Nan::To<float>(info[0]) to info[0].FloatValue()
-  [/Nan::To<float>\((.+?)\)/g, '$1.As<Napi::Number>().FloatValue()'],
-  // ex. Nan::To<double>(info[0]) to info[0].DoubleValue()
-  [/Nan::To<double>\((.+?)\)/g, '$1.As<Napi::Number>().DoubleValue()'],
+  [ /\.IsInt32\(\)/g, '.IsNumber()' ],
+  [ /->IsInt32\(\)/g, '.IsNumber()' ],
 
-  [/Nan::New\((\w+)\)->HasInstance\((\w+)\)/g, '$2.InstanceOf($1.Value())'],
 
-  [/Nan::Get\(([^)]+),\s*/gm, '($1).Get('],
-  [/\.Get\([\s|\\]*Nan::New\(([^)]+)\)\)/gm, '.Get($1)'],
+  [ /(.+?)->BooleanValue()/g, '$1.As<Napi::Boolean>().BooleanValue()' ],
+  [ /(.+?)->Int32Value()/g, '$1.As<Napi::Number>().Int32Value()' ],
+  [ /(.+?)->Uint32Value()/g, '$1.As<Napi::Number>().Uint32Value()' ],
+  [ /(.+?)->IntegerValue()/g, '$1.As<Napi::Number>().Int64Value()' ],
+  [ /(.+?)->NumberValue()/g, '$1.As<Napi::Number>().DoubleValue()' ],
 
-  [/Nan::Set\(([^,]+),\s*/gm, '($1).Set('],
-  [/\.Set\([\s|\\]*Nan::New\(([^)]+)\),/gm, '.Set($1,'],
+  // ex. Nan::To<bool>(info[0]) to info[0].Value()
+  [ /Nan::To<(Boolean|String|Number|Object|Array|Symbol|Function)>\((.+?)\)/g, '$2.To<Napi::$1>()' ],
+  [ /Nan::To<v8::(Boolean|String|Number|Object|Array|Symbol|Function)>\((.+?)\)/g, '$2.To<Napi::$1>()' ],
+  // ex. Nan::To<bool>(info[0]) to info[0].As<Napi::Boolean>().Value()
+  [ /Nan::To<bool>\((.+?)\)/g, '$1.As<Napi::Boolean>().Value()' ],
+  // ex. Nan::To<int>(info[0]) to info[0].As<Napi::Number>().Int32Value()
+  [ /Nan::To<int>\((.+?)\)/g, '$1.As<Napi::Number>().Int32Value()' ],
+  // ex. Nan::To<int32_t>(info[0]) to info[0].As<Napi::Number>().Int32Value()
+  [ /Nan::To<int32_t>\((.+?)\)/g, '$1.As<Napi::Number>().Int32Value()' ],
+  // ex. Nan::To<uint32_t>(info[0]) to info[0].As<Napi::Number>().Uint32Value()
+  [ /Nan::To<uint32_t>\((.+?)\)/g, '$1.As<Napi::Number>().Uint32Value()' ],
+  // ex. Nan::To<int64_t>(info[0]) to info[0].As<Napi::Number>().Int64Value()
+  [ /Nan::To<int64_t>\((.+?)\)/g, '$1.As<Napi::Number>().Int64Value()' ],
+  // ex. Nan::To<float>(info[0]) to info[0].As<Napi::Number>().FloatValue()
+  [ /Nan::To<float>\((.+?)\)/g, '$1.As<Napi::Number>().FloatValue()' ],
+  // ex. Nan::To<double>(info[0]) to info[0].As<Napi::Number>().DoubleValue()
+  [ /Nan::To<double>\((.+?)\)/g, '$1.As<Napi::Number>().DoubleValue()' ],
+
+  [ /Nan::New\((\w+)\)->HasInstance\((\w+)\)/g, '$2.InstanceOf($1.Value())' ],
+
+  [ /Nan::Get\(([^)]+),\s*/gm, '($1).Get(' ],
```

Comment:
I think `/Nan::Get\(([^,\s]+)\s*,\s*/gm` is better for nested stuff. And maybe add the same thing for `Nan::Has()`